pipeline/setup-runner.py

```diff
@@ -0,0 +1,347 @@
+        # Houdini 19.5 uses vc142, others use vc143
+        vc_version = "vc142" if version.startswith("19.5") else "vc143"
+        houdini_installer = Path(f"C:/Tools/houdini-{version}-installer.exe")
+        houdini_installer.parent.mkdir(parents=True, exist_ok=True)
+
+        download_from_s3(
+            f"houdini/{major_minor}/houdini-{version}-win64-{vc_version}.exe", houdini_installer
+        )
+        verify_checksum(houdini_installer, HOUDINI_CHECKSUMS[version]["windows"])
+
+        print("Starting installation...")
+        result = subprocess.run(
+            [
+                houdini_installer,
+                "/S",
+                "/AcceptEULA=2021-10-13",
+                f"/InstallDir={houdini_dir}",
+                "/InstallHoudini=Yes",
+                "/InstallLicense=No",
+                "/InstallMenus=No",
+            ],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+
+        print(f"Installation exit code: {result.returncode}")
+        if result.stdout:
+            print(f"Installation output: {result.stdout}")
+        if result.stderr:
+            print(f"Installation errors: {result.stderr}")
+
+        # Verify installation
+        print(f"Checking installation at {houdini_dir}")
+        if houdini_dir.exists():
+            print("Directory exists, contents:")
+            run(["cmd", "/c", "dir", str(houdini_dir)], check=False)
+            hython_exe = houdini_dir / "bin" / "hython.exe"
+            if hython_exe.exists():
+                print(f"SUCCESS: hython.exe found at {hython_exe}")
+                houdini_marker.touch()  # Create marker on success
+            else:
+                print(f"ERROR: hython.exe NOT found at {hython_exe}")
+        else:
+            print(f"ERROR: Installation directory {houdini_dir} does not exist")
+
+        houdini_installer.unlink(missing_ok=True)
+
+    print("Installing Houdini submitter...")
+    for version in HOUDINI_VERSIONS:
+        major_minor = ".".join(version.split(".")[:2])
+        print(f"Installing submitter for Houdini {version} (major_minor: {major_minor})")
+        result = subprocess.run(
+            ["hatch", "run", "install", "--houdini-version", major_minor],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        print(f"Hatch install exit code: {result.returncode}")
+        if result.stdout:
+            print(f"Hatch install stdout: {result.stdout}")
+        if result.stderr:
+            print(f"Hatch install stderr: {result.stderr}")
+        if result.returncode != 0:
+            print(f"ERROR: Hatch install failed for Houdini {version}")
+            sys.exit(result.returncode)
+
+        # Install pywin32 into Houdini's Python
+        houdini_dir = Path(f"C:/Tools/houdini-{version}")
+
+        # Map Houdini versions to Python versions
+        if version.startswith("19.5"):
```

Comment:
nit: maybe we should add a trailing `.`


```suggestion
        if version.startswith("19.5."):
```

same for other conditions below

Reply:
Added data structure for versions.